File: next/courses/courses-hub/dto/LessonGetResponse.ts
import { TimeUnit } from "@/types/TimeUnit";
import { Survey } from "../../types/Survey";

export interface LessonGetResponse {
    _id: string;
    title: string;
    time?: TimeUnit;
    imageUrl?: string;
    additionalInfo?: string;
    isVisible?: boolean;

    videoId?: string;
    synopsisText?: string; // TODO: stuff
    surveyText?: string;
    survey?: Survey; // TODO: only id for small response

    isMandatory?: boolean;
}

File: kotlin/admin/src/main/kotlin/eduflow/admin/course/dto/lesson/LessonGetResponse.kt
package eduflow.admin.course.dto.lesson

import eduflow.admin.course.models.survey.CourseSurveyModel
import eduflow.course.CourseLesson
import java.util.*

data class LessonGetResponse(
    override val _id: String,
    override val courseId: String?,
    override val createdAt: Date,
    override val imageUrl: String?,
    override val isDraft: Boolean?,
    override val isVisible: Boolean,
    override val sectionId: String?,
    override val surveyText: String?,
    override val synopsisText: String?,
    val survey: CourseSurveyModel? = null,
    override val time: String?,
    override val title: String,
    override val updatedAt: Date,
    override val videoId: String?
) : CourseLesson
